opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1422,26 +1423,41 @@ void BlackoilWellModelGeneric::initInjMult() {
 }
 
 
-void BlackoilWellModelGeneric::updateFiltrationParticleVolume(const double dt, const size_t water_index) {
+void BlackoilWellModelGeneric::updateFiltrationParticleVolume(const double dt,
+                                                              const size_t water_index)
+{
     for (auto& well : this->well_container_generic_) {
         if (well->isInjector() && well->wellEcl().getFilterConc() > 0.) {
-            auto &values =  this->filtration_particle_volume_[well->name()];
-            const auto& ws = this->wellState().well(well->indexOfWell());
-            if (values.empty()) {
-                values.assign(ws.perf_data.size(), 0.); // initializing to be zero
+            auto it = this->filter_cake_.find(well->name());
+            if (it == this->filter_cake_.end()) {
+                this->filter_cake_.insert({well->name(), WellFilterCake(*well)});
+                it = this->filter_cake_.find(well->name());
             }
-            well->updateFiltrationParticleVolume(dt, water_index, this->wellState(), values);
+            it->second.updateFiltrationParticleVolume(dt, water_index, this->wellState());
```

Comment:
I think you could probably use `unordered_map<>::emplace()` here:
```C++
auto [it, inserted] = this->filter_cake_
    .emplace(std::piecewise_construct,
             std::forward_as_tuple(well->name()),
             std::forward_as_tuple(*well));
```
which will insert a new node (`it`), and return `inserted = true`, if `well->name()` does not exist, or return the existing node (`it`) and `inserted = false` otherwise.  That way you don't have to look up `well->name()` twice.

Reply:
done.